dev/com.ibm.ws.security.openidconnect.client/src/com/ibm/ws/security/openidconnect/client/internal/AccessTokenAuthenticator.java

```diff
@@ -1015,6 +1010,59 @@ private static boolean isBearerToken(String hdrValue) {
         return hdrValue != null && hdrValue.startsWith(BEARER_SCHEME);
     }
 
+    public boolean canUseIssuerAsSelectorForInboundPropagation(HttpServletRequest req, OidcClientConfig clientConfig) {
+        boolean result = false;
+
+        // If inbound propagation is not enabled or "disableIssChecking" is set, return false
+        if (!clientConfig.isInboundPropagationEnabled() || clientConfig.disableIssChecking()) {
+            return false;
+        }
+
+        String issuer = getIssuer(req, clientConfig);
+        if (issuer != null) {
+            String issuers = null;
+            if (issuer.isEmpty()
+                    || ((issuers = getIssuerIdentifier(clientConfig)) == null)
+                    || notContains(issuers, issuer)) {
+            } else {
+                result = true;
+            }
+        }
+
+        return result;
+    }
+
+    @FFDCIgnore({ IOException.class })
+    @Trivial
+    private String getIssuer(HttpServletRequest req, OidcClientConfig clientConfig) {
+        String issuer = null;
+        String accessToken = getAccessToken(req, clientConfig);
+
+        if (accessToken != null && isTokenJWT(accessToken)) {
+            try {
+                JSONObject payload = JSONObject.parse(JsonTokenUtil.fromBase64ToJsonString(accessToken.split("\\.")[1]));
```

Comment:
Note that this will only work for JWS tokens. If we want to support JWEs as well this could be updated to something like
```suggestion
                String payloadString;
                if (JweHelper.isJwe(accessToken)) {
                    payloadString = JweHelper.extractPayloadFromJweToken(accessToken, clientConfig, null);
                } else {
                    payloadString = JsonTokenUtil.fromBase64ToJsonString(accessToken.split("\\.")[1])
                }
                JSONObject payload = JSONObject.parse(payloadString);
```